### DISCUSSION TITLE
How can make a conditional on:click argument?

### DISCUSSION THREAD

**Discussion by 205g0** [sveltejs/kit#8541] [Q&A]
i.e. I have a `<Component on:click={() => console.log() />` which is called in follwing div...
```typescript
<div on:click> // <= I want to make this `on:click` arg conditional
  <Component />
</div>
```
So basically, I want want to make `on:click` on the call level optional but once I pass something it will think it's the actual `on:click` function. Hope I've been clear.  Thanks!

**Comment by NormandoHall** [comment]
Why not wrap in an `{#if}` block?

```svelte
{#if variableToCheck}
  <div on:click> // <= I want to make this `on:click` arg conditional
    <Component />
  </div>
{/if}
```

Or you can check inside the component the conditional variable to handle or not

**Reply by 205g0** [reply]
Wouldn't work for my use case (check my other reply) but thanks!

**Comment by teemingc** [comment]
Do you meant to conditionally check and add the event listener? The `on:click` directive for svelte isn't optional at the moment.
(see a list of event directive modifiers [here](https://svelte.dev/docs#template-syntax-element-directives-on-eventname)).

```svelte
<script script="ts">
  import { onMount } from 'svelte';

  let element: HTMLElement;

  onMount(() => {
    if (conditional) {
      element.addEventListener('click', () => {
        // ...
      })
    }
  })
</script>

<div bind:this={element}>
  <Component />
</div>
```

Unless you want to do the conditional check in the function itself where the event listener is already added.

```svelte
<div on:click={() => {
    if (conditional) {
       // ...
    }
  }}>
  <Component />
</div>
```

**Reply by 205g0** [reply]
First idea is good for my use case (which is disabled button elements (not `<button />`) which should not process the click).

**Comment by neopointer** [comment]
In svelte 5 I did like this, but not sure if that's what you meant.

```html
<input onclick={inputOnClick ?? undefined} />
```

**Reply by mateothegreat** [reply]
It'd be easy to throw the problem at an attachment!

```svelte
<script>
    let isActive = $state(true);

    const clickable = (node) => {
        const handleClick = () => {
            alert('Clicked!');
        }

        // Track current active state INSIDE:
        $effect(() => {
            if (isActive) {
                node.addEventListener('click', handleClick);
            } else {
                node.removeEventListener('click', handleClick);
            }
        });

        return {
            destroy() {
                node.removeEventListener('click', handleClick);
            }
        };
    };

    const toggle = () => {
        isActive = !isActive;
    };
</script>

<button onclick={toggle}>
    Toggle clickable (currently {isActive ? 'on' : 'off'})
</button>

<div use:clickable>
    Click me? (I'm {isActive})
</div>
```

See https://svelte.dev/playground/ea5be303525349fa88820fd1c1c2e11f?version=5.38.2.

**Reply by brunnerh** [reply]
(That is an "action", not an "attachment" which use `@attach`.)
